### ISSUE TITLE
go-golden should support pretty-printed golden files

### ISSUE THREAD

**Issue by vsinha** [jimeh/go-golden#4]
Hi jimeh, have you thought about supporting something like [json.Indent](https://pkg.go.dev/encoding/json#Indent) for updated golden files? It would make them cleaner to read manually, which I think would be a nice thing to have.

Excited about this library!

**Comment by jimeh** [comment]
I believe what is written to golden files is actually outside the scope of the golden package itself, as it just reads/writes byte slices.

But I have been on and off working on a companion package that has assertion helpers for JSON, YAML, and XML. Along with a means for supplying custom marshal/unmarshal functions for any kind of wire format.

I can't promise a timeline, but I want to get it finished "soon", as it would replace and simplify nearly all of my own current uses of the golden package.

**Comment by vsinha** [comment]
That makes sense. I wrote up [a helper function](https://github.com/vsinha/launchdata/blob/ea97110510ed79e0edcdee9151894f961ca7aa16/lib/test_helpers.go#L18) for my json-specific application, is this the kind of thing you had in mind?

Just getting started with Go / getting a lay of the land wrt libraries and best practices

**Comment by jimeh** [comment]
Yep, that's roughly the kind of thing I'm doing myself for structs that serialized to JSON/YAML/XML.

And the companion package I'm working on will do that, but lots of flexibility :)
